src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
Both Pear and Artifact repository lack this method even though they both accept and work with $repoConfig. @alcohol Could this perhaps be implemented in Composer with a PR? Because at the moment I don't know how to get hold of repository url.